core/network/types/sync_clients_set.hpp

```diff
@@ -16,7 +16,7 @@ namespace kagome::network {
    * Keeps all known Sync clients
    */
   struct SyncClientsSet {
-    std::unordered_set<std::shared_ptr<SyncProtocolClient>> clients;
+    std::vector<std::shared_ptr<SyncProtocolClient>> clients;
```

Comment:
Is it to enable getting client by index? What about performance decrease?

Reply:
it is not significant